backend/layers/business/business.py

```diff
@@ -931,6 +938,14 @@ def update_dataset_artifact(self, artifact_id: DatasetArtifactId, artifact_uri:
         """
         self.database_provider.update_dataset_artifact(artifact_id, artifact_uri)
 
+    def add_artifact_to_dataset_version(
+        self, dataset_version_id: DatasetVersionId, artifact_id: DatasetArtifactId
+    ) -> None:
+        """
+        Adds an artifact to a dataset version
+        """
+        self.database_provider.add_artifact_to_dataset_version(dataset_version_id, artifact_id)
+
```

Comment:
I personally would not define this method and would instead just call `.database_provider.add_artifact_to_dataset_version`